### PR TITLE
Port PRs #1148 and #1149 to master

### PR DESCRIPTION
**Pull request by fizyk20**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
Thanks for the pull request, and welcome! The MaidSafe team is excited to review your changes, and you should hear from @afck (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTOR.md](../blob/master/CONTRIBUTOR) for more information.

File: examples/ci_test.rs [hunk 1]
```diff
@@ -341,15 +344,11 @@ fn main() {
...
             cvar.notify_one();
         }
-    } else {
-        if let Some(log_file) = args.flag_output {
-            unwrap_result!(maidsafe_utilities::log::init_to_file(false, log_file, true));
-        } else {
-            unwrap_result!(maidsafe_utilities::log::init(false));
-        }
+    } else if let Some(log_file) = args.flag_output {
+        unwrap!(maidsafe_utilities::log::init_to_file(false, log_file, true));
```
Comment (afck) [inline examples/ci_test.rs]: This should probably be left as it is, just with `unwrap_result!` replaced by `unwrap!`.

File: examples/utils/example_client.rs [hunk 2]
```diff
@@ -49,7 +49,7 @@ impl ExampleClient {
         let sign_keys = crypto::sign::gen_keypair();
         let encrypt_keys = crypto::box_::gen_keypair();
         let full_id = FullId::with_keys(encrypt_keys.clone(), sign_keys.clone());
-        let mut routing_client;
+        let routing_client = unwrap!(Client::new(sender, Some(full_id)));
```
Comment (afck) [inline examples/utils/example_client.rs]: This line should also be left as it is - on master, `routing_client` is initialised in the loop below.

File: examples/utils/example_node.rs [hunk 3]
```diff
@@ -81,7 +81,7 @@ impl ExampleNode {
                 }
                 Event::RestartRequired => {
                     info!("{} Received RestartRequired event", self.get_debug_name());
-                    self.node = unwrap_result!(Node::builder().create(self.sender.clone()));
+                    let new_node = unwrap!(Node::builder().create(self.sender.clone()));
```
Comment (afck) [inline examples/utils/example_node.rs]: Should probably still be `self.node =`.

This file has changed considerably on master and it shouldn't be overwritten with the version from the stable branch. Just change all the `unwrap_result!` and `unwrap_option!` to `unwrap!`.

File: src/message_filter.rs [hunk 4]
```diff
@@ -15,6 +15,7 @@
 // Please review the Licences for the specific language governing permissions and limitations
 // relating to use of the SAFE Network Software.
 
+#![allow(deprecated)]
```
Comment (Fraser999) [inline src/message_filter.rs]: When we disable lints, we normally try and keep the affected scope as small as possible (unless that means duplicating the macro a silly number of times or it is needed for a majority of the functions in a given file), and it's good to add a todo if it's really meant as a temporary fix.

Comment (fizyk20) [inline src/message_filter.rs]: I've actually tried to mark only the line importing SipHasher, but apparently it doesn't work that way. I didn't have any better idea for how to do this - should I add a todo, then?

File: examples/utils/example_node.rs (hunk 3 above)
Comment (fizyk20) [inline examples/utils/example_node.rs]: I've tried to keep the merged changes to just the unwrap!s, but I must have missed a few (especially those large blocks... I'm embarrassed by those :p). Thanks!

File: examples/ci_test.rs (hunk 1 above)
Comment (fizyk20) [inline examples/ci_test.rs]: Right, corrected.

File: examples/utils/example_client.rs (hunk 2 above)
Comment (fizyk20) [inline examples/utils/example_client.rs]: Corrected as well :)

**Pull request merged**